src/path.c

```diff
@@ -56,7 +56,7 @@ alloc_array(Py_ssize_t count)
         PyErr_NoMemory();
         return NULL;
     }
-    if (count > (SIZE_MAX / (2 * sizeof(double))) - 1 ) {
+    if ((unsigned long long)count > (SIZE_MAX / (2 * sizeof(double))) - 1 ) {
```

Comment:
Immediately before this, [if `count < 0` the code `return`s](https://github.com/python-pillow/Pillow/blob/f15c64dbb667190b81fbe86a67dda67200a601cd/src/path.c#L55-L58), so it is safe to cast to unsigned here.